kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -359,53 +386,148 @@ protected ProcurementCardDefault retrieveProcurementCardDefault(String creditCar
+        List<ProcurementCardTargetAccountingLine> procurementCardTargetAccountingLines = (List<ProcurementCardTargetAccountingLine>) pcardDocument.getTargetAccountingLines();
+        for (ProcurementCardTargetAccountingLine procurementCardTargetAccountingLine : procurementCardTargetAccountingLines) {
+            if (procurementCardTargetAccountingLine.getAccountNumber() != null && accountValues.contains(procurementCardTargetAccountingLine.getAccountNumber())) {
+                return true;
+            }
+
+            if (procurementCardTargetAccountingLine.getObjectCode() != null && objectSubTypeValues.contains(procurementCardTargetAccountingLine.getObjectCode().getFinancialObjectSubTypeCode())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * blanket approve document in a transaction. This method was created so each document
+     * is handled in its own transaction. This allows validation errors to be thrown without
+     * rolling back the entire process.
+     *
+     * @param pcardDocument
+     * @param accountList
+     * @param objectSubTypeList
+     * @param autoApproveNumberDays
+     * @throws WorkflowException
+     */
+    private String blanketApproveDocument(ProcurementCardDocument pcardDocument, Timestamp currentDate) throws WorkflowException {
+        StringBuilder errorMessage = new StringBuilder();
+
+        // update document description to reflect the auto approval
+        if (pcardDocument.getDocumentHeader().getDocumentDescription().startsWith("FY ")) {
+            pcardDocument.getDocumentHeader().setDocumentDescription(pcardDocument.getDocumentHeader().getDocumentDescription().substring(0, 7) + " Auto Approve " + dateTimeService.toDateTimeString(currentDate));
+        } else {
+            pcardDocument.getDocumentHeader().setDocumentDescription("Auto Approved On " + dateTimeService.toDateTimeString(currentDate) + ".");
+        }
+        LOG.info("Auto approving document # " + pcardDocument.getDocumentHeader().getDocumentNumber());
+        pcardDocument.setAutoApprovedIndicator(true);
+
+        try {
+            documentService.blanketApproveDocument(pcardDocument, "", null);
+        } catch (ValidationException e) {
+            LOG.error("Error auto approving document # " + pcardDocument.getDocumentNumber() + " " + e.getMessage(), e);
+            errorMessage.append(pcardDocument.getDocumentNumber() + "\n");
+            errorMessage.append(e.getMessage() + "\n");
+            // grab and log all error messages that would have been displayed to the user.
+
+            for (Map.Entry<String, AutoPopulatingList<ErrorMessage>> entry : GlobalVariables.getMessageMap().getErrorMessages().entrySet()) {
+                for (ErrorMessage em : entry.getValue()) {
+                    errorMessage.append(expandErrorString(em.getErrorKey(), em.getMessageParameters()) + "\n");
                 }
-            } catch (WorkflowException e) {
```

Comment:
The WorkflowException for this is being caught and handled at line 418. This is being "removed" and "added" above because the white-spacing is slightly different (different braces level). This method *should* throw an exception if there's a problem in handling this during the workflow, and be caught within the for loop of autoApproveProcurementCardDocuments()